src/agent.spec.ts

```diff
@@ -1,74 +1,173 @@
-          alertId: "FORTA-1",
-          severity: FindingSeverity.Low,
+        expect.objectContaining({
+          name: "Nethermind Bot Deployment Detector",
+          description: "Detects Bots Deployed by Nethermind",
+          alertId: "FORTA-123",
+          severity: FindingSeverity.Info,
           type: FindingType.Info,
           metadata: {
-            to: mockTetherTransferEvent.args.to,
-            from: mockTetherTransferEvent.args.from,
+            agentId:mockTxEventOne.agentId.toString(),
+            owner:mockTxEventOne.owner,
+            metadata: mockTxEventOne.metadata,
+            chainIds:mockTxEventOne.chainIds[0],
           },
         }),
       ]);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledTimes(1);
-      expect(mockTxEvent.filterLog).toHaveBeenCalledWith(
-        ERC20_TRANSFER_EVENT,
-        TETHER_ADDRESS
-      );
     });
+
+  //multiple bot deployments case
+  it("returns a finding if there is multiple bot deployments", async () => {
+    mockTxEvent = new TestTransactionEvent()
+    .setFrom(BOT_DEPLOYER_ADDRESS)
+    .setTo(FORTA_BOT_REGISTRY)
+    .addTraces({
+      function: proxyInterface.getFunction("newAgent"),
+      to: FORTA_BOT_REGISTRY, 
+      from: BOT_DEPLOYER_ADDRESS,
+      arguments: [
+        mockTxEventOne.agentId,
+        mockTxEventOne.owner,
+        mockTxEventOne.metadata,
+        [BigNumber.from(mockTxEventOne.chainIds[0])],
+      ],
+    })
+    .addTraces({
+      function: proxyInterface.getFunction("newAgent"),
+      to: FORTA_BOT_REGISTRY, 
+      from: BOT_DEPLOYER_ADDRESS,
+      arguments: [
+        mockTxEventTwo.agentId,
+        mockTxEventTwo.owner,
+        mockTxEventTwo.metadata,
+        [BigNumber.from(mockTxEventTwo.chainIds[0])],
+      ],
+    })
+    ;
+
+    findings = await handleTransaction(mockTxEvent);
+
+   //make sure findings is equal to the mock transactions we created
+    expect(findings).toStrictEqual([
+      expect.objectContaining({
+        name: "Nethermind Bot Deployment Detector",
+        description: "Detects Bots Deployed by Nethermind",
+        alertId: "FORTA-123",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        metadata: {
+          agentId:mockTxEventOne.agentId.toString(),
+          owner:mockTxEventOne.owner,
+          metadata: mockTxEventOne.metadata,
+          chainIds:mockTxEventOne.chainIds[0],
+        },
+      }),
+      expect.objectContaining({
+        name: "Nethermind Bot Deployment Detector",
+        description: "Detects Bots Deployed by Nethermind",
+        alertId: "FORTA-123",
+        severity: FindingSeverity.Info,
+        type: FindingType.Info,
+        metadata: {
+          agentId:mockTxEventTwo.agentId.toString(),
+          owner:mockTxEventTwo.owner,
+          metadata: mockTxEventTwo.metadata,
+          chainIds:mockTxEventTwo.chainIds[0],
+        },
+      })
+      ,
+    ]);
+  });
+
```

Comment:
I think it would be good to add one more test case, and that is a call to a `newAgent(...)` function that is NOT in the Forta contract. Want to make sure your bot does not create alerts in for bot (fka agents) created for another contract.